pilot/pkg/bootstrap/server.go

```diff
@@ -892,6 +893,7 @@ func (s *Server) initMemoryRegistry(serviceControllers *aggregate.Controller) {
 func (s *Server) initDiscoveryService(args *PilotArgs) error {
 	environment := &model.Environment{
 		Mesh:             s.mesh,
+		MeshUID:          args.MeshUID,
```

Comment:
See above - I don't think we need to pass it around and copy it in many places ( which is an artifact of the CLI passing ).